File: android/GestionareProduse/app/src/main/java/app/gestionareproduse/products/domain/Product.kt
package app.gestionareproduse.products.domain

import androidx.room.Entity
import androidx.room.ForeignKey
import androidx.room.PrimaryKey
import androidx.room.TypeConverters
import app.gestionareproduse.chooseWarehouseScreen.domain.Warehouse
import app.gestionareproduse.utils.DateConverter
import java.util.*

@Entity(tableName = "products_table",
        foreignKeys = arrayOf(ForeignKey(entity = Warehouse::class,
                            parentColumns = arrayOf("id"),
                            childColumns = arrayOf("warehouseId"),
                            onDelete = ForeignKey.CASCADE,
                            onUpdate = ForeignKey.CASCADE)))
@TypeConverters(DateConverter::class)
data class Product(
    @PrimaryKey(autoGenerate = false) var id: Long? = null,
    var name: String,
    var brand: String,
    var price: Double,
    var isPerUnit: Boolean,
    var expirationDate: Date,
    var isRefrigerated: Boolean,
    var image: String,
    var warehouseId: Long,
    var isUploaded : Boolean? = null
)

File: flutter/gestionare_produse/lib/domain/product.dart
class Product{
  late int id;
  late String name;
  late String brand;
  late double price;
  late bool isPerUnit;
  late DateTime expirationDate;
  late bool isRefrigerated;
  late String image;
  late int warehouseId;

  Product({required this.id, required this.name, required this.brand, required this.price, required this.isPerUnit,
      required this.expirationDate, required this.isRefrigerated, required this.image, required this.warehouseId});
}